Develop3

Igor-Belov opened the pull request:
UI тесты для проверки https://stellarburgers.nomoreparties.site/
Реализована последовательная прогонка тестов в двух брахерах (хром и яндекс)
Реализована очистка тестовых данный. Токен авторизации получается в API и передается в браузер.
Подключен аллюр.

Igor-Belov committed:
Сделал тесты SignUp и служебный класс API

Igor-Belov committed:
Запуск через два браузера.

Igor-Belov committed:
Написал тесты на авторизацию с разных точек входа.

Igor-Belov committed:
Написал все тесты. Немного причесал. Приложил аллюр отчет.

egorvas commented on src/test/java/site/nomoreparties/stellarburgers/AccountPageTest.java:
```diff
@@ -0,0 +1,142 @@
+    public static Collection<Object[]> browsers() {
+        return Arrays.asList(new Object[][]{
+                {"yandex"},
+                {"chrome"}
```
**Можно лучше:** выносить браузер в параметры теста не очень хорошая идея. На практике тебе зачастую захочется запустить тесты на определенном браузере или стенды с разными браузерами могут быть в разных местах. Поэтому лучше выносить браузер в параметр запуска, например `mvn verify -Dbrowser=chrome`, чтоб иметь возможность запустить тесты с определенным браузером. А параметры теста оставить для пользовательских данных.

egorvas commented on src/test/java/site/nomoreparties/stellarburgers/AccountPageTest.java:
```diff
@@ -0,0 +1,142 @@
+        } else if (browserName.equals("yandex")) {
+            WebDriverManager.chromedriver().driverVersion("132.0.6834.0").setup();
+            ChromeOptions options = new ChromeOptions();
+            options.setBinary("C:/Users/belov/AppData/Local/Yandex/YandexBrowser/Application/browser.exe");
```
**Нужно исправить:** я не могу принять работу с захардкоженным путем драйвера. Ты можешь использовать папку resources, или передавать путь через переменную среды или сделать конфиг, но хардкод делать нельзя.

Igor-Belov replied:
Исправлено через переменную среды YANDEX_BROWSER_PATH.

egorvas commented on src/test/java/site/nomoreparties/stellarburgers/AccountPageTest.java:
```diff
@@ -0,0 +1,142 @@
+
+        // Создание пользователя для тестов
+        email = LocalDateTime.now() + "@example.com";
+        password = "P@ssword123";
```
**Можно лучше:** Вместо того чтобы самому придумывать тестовые строки я рекомендую использовать модули для генерации данных, например java faker: https://github.com/DiUS/java-faker
С его помощью ты сможешь убрать большую часть хардкода из твоих тестов.

Igor-Belov replied:
Да, faker классная штука, но мне кажется, что в данном случае удобнее использовать даты в имени (понятно когда именно был тест) и фиксированный пароль (что бы зайти под этим пользаком на сайт или постман нужно только поменять имя).  Name же кодирует кто именно автор пользователя (типа Auto_test_Belov). Вот например заполнить таблицу тестовыми данными - тут герои из Гарри Поттера отлично подойдут)

egorvas commented on src/test/java/site/nomoreparties/stellarburgers/ConstructorTest.java:
```diff
@@ -0,0 +1,92 @@
+    @Before
+    public void setUp() {
+        // Настройка драйвера в зависимости от выбранного браузера
+        if (browserName.equals("chrome")) {
```
**Можно лучше:** конфигурацию браузера лучше вынести в общее место для всех тестов

Igor-Belov replied:
Создал класс DriverFactory

Igor-Belov committed:
Поправил:
1. выбор браузера через mvn test -Dbrowser=chrome, а не параметризированно
2.Путь до яндекс браузера перенесен в переменную среды ОС YANDEX_BROWSER_PATH
3. Вынес настройку драйвера в DriverFactory
